### ISSUE TITLE
Wrong means, medians

### ISSUE THREAD

**Issue by oleksii-nikolaienko** [benjaminrich/table1#96]
Hi,
I get a very strange output for means, medians and max values in this example:

`table1::table1(~ x | y, data.frame(x=c(2000:2009), y="Y"))`

![image](https://user-images.githubusercontent.com/36014136/216328700-d3b35f89-f2e5-43b5-91b6-98d3922ab7f9.png)

While the following works correctly:

`table1::table1(~ x | y, data.frame(x=c(0:9), y="Y"))`

![image](https://user-images.githubusercontent.com/36014136/216328782-e77b6199-e5e4-42c8-bb98-b8151fbf4ad8.png)

It seems to have something to do with rounding when I try to use other ranges of x: `c(10, 19)`, `c(100, 109)`, `c(1000, 1009)`.
I think the output is very wrong when it comes to years (as I have in the first example)

**Comment by benjaminrich** [comment]
Please have a look at #24.

There is nothing wrong with the calculations, just increase the number of significant digits to 5 (default is 3).

```r
table1::table1(~ x | y, data.frame(x=c(2000:2009), y="Y"), digits=5)
```

(Aside: While it seems strange to me that you would want to take the the mean of years, I guess you know what you are doing)

**Comment by oleksii-nikolaienko** [comment]
Yes, sorry for that, I didn't know default value for `digits=3`. Can see it specified in `signif_pad()`, but not in `table1()`.
On my issue: in this case I wanted to know median and min/max, and instead of `2005 [2001, 2009]` got `2010 [2000, 2010]`. And pasted it right into my manuscript without checking with `summary()`. So while I understand that there is no bug here, it is still a bit misleading, especially given the different default for number of digits... Made me think how many times I could make the same mistake before...

**Comment by oleksii-nikolaienko** [comment]
I get your point on the default values, but some people get confused. Would it be possible to specifically explain this better in the `table1()` help page?

**Comment by benjaminrich** [comment]
Thanks. I will take in under consideration for the next release.

**Comment by ffmed** [comment]
Hey, 
in medical trials you do encounter the median year for treatment a vs b occassionally. 
A simple fix I use is to just customize the digits depending on the value:


```
 render_cont.ff <- function(x, name, data2, ...) {

  MIN <- min(x, na.rm = T)
  MAX <- max(x, na.rm = T)
  median <- median(x, na.rm = T)
  Q1 <- quantile(x, 0.25, na.rm = T)
  Q3 <- quantile(x, 0.75, na.rm = T)
  N = length(x) - sum(is.na(x))
  if (median >1000)
  {
    sprintf("%s (%s, %s)",
            signif_pad(median, 4, big.mark=","),
            signif_pad(Q1,    4, big.mark=","),
            signif_pad(Q3,    4, big.mark=","))
  }
  else
  {
    sprintf("%s (%s, %s)",
            signif_pad(median, 3, big.mark=","),
            signif_pad(Q1,    3, big.mark=","),
            signif_pad(Q3,    3, big.mark=","))
  }  
 
}

```

**Comment by benjaminrich** [comment]
The package is flexible enough to do what you want in most cases, but you might have to do some work to get exactly what you want if it's not the default. If you post an example of exactly what you would like to achieve, I would be happy to work through it with you.
